src/libs/Network/PersistedRequestsQueue.js

```diff
@@ -0,0 +1,96 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import * as PersistedRequests from '../actions/PersistedRequests';
+import * as NetworkStore from './NetworkStore';
+import * as NetworkEvents from './NetworkEvents';
+import CONST from '../../CONST';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as ActiveClientManager from '../ActiveClientManager';
+import processRequest from './processRequest';
+
+let persistedRequestsQueueRunning = false;
+
+/**
+ * This method will get any persisted requests and fire them off in parallel to retry them.
+ * If we get any jsonCode besides 407 the request is a success. It doesn't make sense to
+ * continually retry things that have returned a response. However, we can retry any requests
+ * with known networking errors like "Failed to fetch".
+ *
+ * @returns {Promise}
+ */
+function process() {
+    const persistedRequests = PersistedRequests.getAll();
+
+    // This sanity check is also a recursion exit point
+    if (NetworkStore.getIsOffline() || _.isEmpty(persistedRequests)) {
+        return Promise.resolve();
+    }
+
+    const tasks = _.map(persistedRequests, request => processRequest(request)
+        .then((response) => {
+            if (response.jsonCode === CONST.JSON_CODE.NOT_AUTHENTICATED) {
+                NetworkEvents.getLogger().info('Persisted optimistic request needs authentication');
+            } else {
+                NetworkEvents.getLogger().info('Persisted optimistic request returned a valid jsonCode. Not retrying.');
+            }
+            NetworkEvents.onResponse(request, response);
+            PersistedRequests.remove(request);
+        })
+        .catch((error) => {
+            // If we are catching a known network error like "Failed to fetch" allow this request to be retried if we have retries left
```

Comment:
```suggestion
            // If we are catching a known network error like "Failed to fetch", allow this request to be retried if we have retries left
```

Reply:
Want me to fix the use of passive voice as well :trollface: